src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbByTable.java

```diff
@@ -215,9 +217,22 @@ public QueryResult query(QueryRequest queryRequest) {
      */
     @Override
     public ScanResult scan(ScanRequest scanRequest) {
-        scanRequest = scanRequest.clone();
-        scanRequest.withTableName(buildPrefixedTableName(scanRequest.getTableName()));
-        return getAmazonDynamoDb().scan(scanRequest);
+        if (getMtContext().getContextOpt().isEmpty()) {
+            Preconditions.checkArgument(isMtTable(scanRequest.getTableName()));
+            ScanResult result =  getAmazonDynamoDb().scan(scanRequest);
+            String[] tenantTable = getTenantAndTableName(scanRequest.getTableName());
+            List<String> tenants =new ArrayList(result.getCount());
+            List<String> tables = new ArrayList(result.getCount());
```

Comment:
somewhat silly to build a huge list of the same items, but I figure, its better to stay compatible across byTable or bySharedTable with the same MtScanResult object def